src/js/modules/main/main.js

```diff
@@ -1,86 +1,84 @@
+    this.submit.model.set('visible', collection.length > 0);
+  }
+}
+
+function handleMapVisibleChange(visible) {
+  if (this.app.getState() !== Const.State.VIEWER) {
+    this.start.model.set({ visible });
+  }
+}
+
+function handleStartPlanting() {
+  this.$el.toggleClass(IS_PLANTING_CLASS, true);
+  this.start.model.set('visible', false);
+}
+
+function handleStateChange(state) {
+  this.$el.children().attr('data-state', state);
+}
+
+/**
+ * @public
+ */
 export default View.extend({
   toolbox: null,
   map: null,
   className: 'plantingjs-container',
-  template: require('./main.hbs'),
   $proxy: null,
 
   initialize() {
     this.render();
     this.$proxy = this.$el.children();
-    this.submit = new Button(SUBMIT_BUTTON_INIT_VALUES);
-    this.submit.on('click', this.onClickSubmit, this);
-    this.session().objects().on('add remove', (model, collection) => {
-      if (this.app.getState() !== Const.State.VIEWER) {
-        const showSubmitButton = collection.length > 0;
-        this.submit.model.set('visible', showSubmitButton);
-      }
-    });
-
-    const { selectPanoMode } = this.app.options;
-
-    if (selectPanoMode) {
-      this.start = new Button(SELECT_PANO_INIT_VALUES);
-      this.start.on('click', () => {
-        const { onSelectPano } = this.app.options;
-        const panoData = this.manifesto()
-          .pick('lat', 'lng', 'pitch', 'heading', 'zoom');
+    this.submit = submitButton({ click: this.onClickSubmit }, this);
+    this.session()
+        .objects()
+        .on('add remove', handlePlantedObjectsChanged, this);
 
-        if (isFunction(onSelectPano)) {
-          onSelectPano(panoData);
-        } else {
-          throw Error('onSelectPano must be a function');
-        }
-      });
+    if (this.app.options.selectPanoMode) {
+      this.start = selectButton({ click: handleSelectPano }, this);
     } else {
-      this.start = new Button(START_BUTTON_INIT_VALUES);
-      this.start.on('click', () => {
-        this.app.trigger(Const.Event.START_PLANTING);
-      });
+      this.start = initButton({ click: handleInitPlanting }, this);
     }
 
     this.$proxy.append(this.submit.$el, this.start.$el);
     this.app
-      .on(Const.Event.VISIBLE_CHANGED, (visible) => {
-        if (this.app.getState() !== Const.State.VIEWER) {
-          this.start.model.set({ visible });
-        }
-      })
-      .on(Const.Event.START_PLANTING, () => {
-        this.$el.toggleClass(IS_PLANTING_CLASS, true);
-        this.start.model.set('visible', false);
-      })
-      .on(Const.Event.STATE_CHANGED, (state) => {
-        this.$el
-          .children().attr('data-state', state);
-      });
+        .on(Const.Event.VISIBLE_CHANGED, handleMapVisibleChange, this)
+        .on(Const.Event.START_PLANTING, handleStartPlanting, this)
+        .on(Const.Event.STATE_CHANGED, handleStateChange, this);
   },
 
   render() {
-    this.$el.html(this.template());
+    this.$el.html(template());
```

Comment:
@magul no, previously we were using `require(<template>)` in Backbone class declaration. I changed it to ES6 import, so now template is in module scope. And I think we don't need template as public member.